sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -136,56 +151,62 @@ public static Read read() {
 
       abstract Builder setInitialPosition(StartingPoint startingPoint);
 
-      abstract Builder setClientProvider(KinesisClientProvider clientProvider);
+      abstract Builder setAWSClientsProvider(AWSClientsProvider clientProvider);
 
       abstract Builder setMaxNumRecords(int maxNumRecords);
 
       abstract Builder setMaxReadTime(Duration maxReadTime);
 
+      abstract Builder setAllowedRecordLateness(Duration allowedRecordLateness);
+
       abstract Read build();
     }
 
     /**
-     * Specify reading from streamName at some initial position.
+     * Specify reading from streamName.
      */
-    public Read from(String streamName, InitialPositionInStream initialPosition) {
-      return toBuilder()
-          .setStreamName(streamName)
-          .setInitialPosition(
-              new StartingPoint(checkNotNull(initialPosition, "initialPosition")))
+    public Read withStreamName(String streamName) {
```

Comment:
This is a backwards-incompatible change to a public API: it will cause user code to stop compiling. Beam has had an API-stable release, so we can no longer make incompatible changes: please revert this one, or make it backwards-compatible. Same applies to other renames and signature changes in this PR.